tools/weather.py

```diff
@@ -0,0 +1,55 @@
+import requests
+import logging
+
+from config import OPENWEATHER_API
+
+base_url = "https://api.openweathermap.org/data/2.5/weather?"
+
+CODE_SUCCESS = "200"
+CODE_INVALID_API_KEY = "401"
+CODE_CITY_NOT_FOUND = "404"
+
+
+class InvalidApiKeyError(ValueError):
+    pass
+
+
+def get_weather(city_name: str):
+    complete_url = base_url + "appid=" + OPENWEATHER_API + \
+        "&q=" + city_name + "&units=metric"
+
+    response = requests.get(complete_url)
+    json_full_object = response.json()
+    logging.debug(str(json_full_object))
+
+    response_code = str(json_full_object["cod"])
+
+    if response_code == CODE_SUCCESS:
+        return __convert_json_to_weather_string(json_full_object)
+    elif response_code == CODE_INVALID_API_KEY:
+        raise InvalidApiKeyError
+    elif response_code == CODE_CITY_NOT_FOUND:
+        return "City " + f"<b>{city_name}</b>" + \
+               " is not found, try a different city name. \n" \
+               "example usage: /weather moscow"
+
+
+def __convert_json_to_weather_string(weather_json):
+    json_main_object = weather_json["main"]
+    current_temperature = json_main_object["temp"]
+    current_pressure = json_main_object["pressure"]
+    current_humidity = json_main_object["humidity"]
+
+    json_weather_object = weather_json["weather"]
+    weather_description = json_weather_object[0]["description"]
+
+    city_name = weather_json["name"]
+    json_system_info = weather_json["sys"]
+    country_code = json_system_info["country"]
+
+    return "City: " + city_name + \
+           "\nCountry code: " + country_code + \
+           "\nCurrent temperature: " + str(current_temperature) + " °C" + \
+           "\nCurrent pressure: " + str(current_pressure) + " hPa" + \
+           "\nHumidity: " + str(current_humidity) + "%" + \
+           "\nDescription: " + str(weather_description)
```

Comment:
`city_name = weather_json["name"]`
this can be changed to `city_name = weather_json.get("name")` to avoid `KeyError` exception, or maybe you can just handle the exception somewhere. (idk if you already did it). or maybe it's a very rare case and you can use ostrich algorithm (jk).

maybe it's more readable this way, and reduces those `+ \` and `str()`, but it's just coding style, does not really matter.

```
def foobar():
    return (
        f"City: {city_name}"
        f"\nCountry code: {country_code}"
        f"\nCurrent temperature: {current_temperature} °C"
        f"\nCurrent pressure: {current_pressure} hPa"
        f"\nHumidity: {current_humidity}%"
        f"\nDescription: {weather_description}"
    )
```

or even reduce `\n`

```
def foobar():
    return "\n".join([
        f"City: {city_name}",
        f"Country code: {country_code}",
        f"Current temperature: {current_temperature} °C",
        f"Current pressure: {current_pressure} hPa",
        f"Humidity: {current_humidity}%",
        f"Description: {weather_description}",
    ])
```